### ISSUE TITLE
Usage documentation examples use obsolete net.ltgt.apt plugin for Gradle

### ISSUE THREAD

**Issue by mjustin** [mapstruct/mapstruct#2346]
The Gradle documentation in the [reference guide](https://mapstruct.org/documentation/stable/reference/html/#_gradle) and [project readme](https://github.com/mapstruct/mapstruct#gradle) both instruct the reader to use the `net.ltgt.apt` plugin.  However, this plugin has [declared itself obsolete](https://github.com/tbroyer/gradle-apt-plugin#gradle-apt-plugin) as of Gradle 5.2:

> The goal of this plugin was to eventually no longer be needed, being superseded by built-in features.
This has become a reality with Gradle 5.2 and IntelliJ IDEA 2019.1. **tl;dr: this plugin is obsolete, don't use it.** If you're using Eclipse though, continue reading.
> [...]
> **If you're using Eclipse, please migrate to the [`com.diffplug.eclipse.apt`](https://plugins.gradle.org/plugin/com.diffplug.eclipse.apt) plugin, which is a (maintained) fork of `net.ltgt.apt-eclipse`.**

In short, unless Eclipse is being used, you can use the built-in Gradle annotation processing functionality.  If Eclipse _is_ being used, it looks like the `com.diffplug.eclipse.apt` plugin should be used instead.

Therefore, the documentation should be updated to use the built-in Gradle annotation processor support, rather than the obsolete plugin.

**Comment by filiphr** [comment]
Thanks for pointing this out. We are going to update the documentation and readme for this.
